synapse/storage/controllers/state.py

```diff
@@ -524,12 +525,53 @@ async def get_current_state_event(
         return state_map.get(key)
 
     async def get_current_hosts_in_room(self, room_id: str) -> List[str]:
-        """Get current hosts in room based on current state."""
+        """Get current hosts in room based on current state.
+
+        Blocks until we have full state for the given room. This only happens for rooms
+        with partial state.
+
+        Returns:
+            A list of hosts in the room, sorted by longest in the room first. (aka.
+            sorted by join with the lowest depth first).
+        """
 
         await self._partial_state_room_tracker.await_full_state(room_id)
 
         return await self.stores.main.get_current_hosts_in_room(room_id)
 
+    async def get_current_hosts_in_room_or_partial_state_approximation(
+        self, room_id: str
+    ) -> Sequence[str]:
```

Comment:
Shouldn't we use `Collection` since `Sequence` implies an order and it's possible we don't have one ?
That would allow to return directly the `set` in my following suggestion.

Reply:
I wanted the method to be interchangeable with `get_current_hosts_in_room`, which means having the same signature and ordering (+/- mutability).